py-polars/src/series/buffers.rs

```diff
@@ -164,3 +323,87 @@ fn get_ptr<T: PolarsNumericType>(ca: &ChunkedArray<T>) -> usize {
     let arr = ca.downcast_iter().next().unwrap();
     arr.values().as_ptr() as usize
 }
+
+unsafe fn from_buffer_impl<T: NativeType>(
+    pointer: usize,
+    length: usize,
+    base: Py<PyAny>,
+) -> Series {
+    let pointer = pointer as *const T;
+    let slice = unsafe { std::slice::from_raw_parts(pointer, length) };
+    let arr = unsafe { arrow::ffi::mmap::slice_and_owner(slice, base) };
+    Series::from_arrow("", arr.to_boxed()).unwrap()
+}
+unsafe fn from_buffer_boolean_impl(
+    pointer: usize,
+    offset: usize,
+    length: usize,
+    base: Py<PyAny>,
+) -> PyResult<Series> {
+    let length_in_bytes = get_boolean_buffer_length_in_bytes(length, offset);
+
+    let pointer = pointer as *const u8;
+    let slice = unsafe { std::slice::from_raw_parts(pointer, length_in_bytes) };
+    let arr_result = unsafe { arrow::ffi::mmap::bitmap_and_owner(slice, offset, length, base) };
+    let arr = arr_result.map_err(PyPolarsErr::from)?;
+    let s = Series::from_arrow("", arr.to_boxed()).unwrap();
+    Ok(s)
+}
+fn get_boolean_buffer_length_in_bytes(length: usize, offset: usize) -> usize {
+    let n_bits = offset + length;
+    let n_bytes = n_bits / 8;
+    let rest = n_bits % 8;
+    if rest == 0 {
+        n_bytes
+    } else {
+        n_bytes + 1
+    }
+}
+
+fn series_to_buffer<T>(s: Series) -> PyResult<Buffer<T::Native>>
+where
+    T: PolarsNumericType,
+{
+    let ca: &ChunkedArray<T> = s.as_any().downcast_ref().unwrap();
```

Comment:
Apparently this doesn't work for float types... any ideas on how to generically implement this for all numeric types? Maybe I should just write a macro for this?